model/flow/transaction_result.go

```diff
@@ -33,3 +33,23 @@ func (te *TransactionResult) Checksum() Identifier {
 
 // TODO(ramtin): add canonical encoding and ID
 type TransactionResults []TransactionResult
+
+// LightTransactionResult represents a TransactionResult, omitting any fields that are prone to
+// non-determinism; i.e. the error message and memory used estimate.
+//
+// While the net causes of a transaction failing are deterministic, the specific error and message
+// propagated back to FVM are prone to bugs resulting in slight variations. Rather than including
+// the error and risking execution forks if an undetected bug is introduced, we simplify it to just
+// a boolean value. This will likely change in the future to include some additional information
+// about the error.
+//
+// Additionally, MemoryUsed is omitted because it is an estimate from the specific execution node,
+// and will vary across nodes.
+type LightTransactionResult struct {
+	// TransactionID is the ID of the transaction this result was emitted from.
+	TransactionID Identifier
+	// Failed is true if the transaction's execution failed resulting in an error, false otherwise.
+	Failed bool
```

Comment:
This might be non-relevant but just a random thought about having a field called `failed` - by default, it will be false, if someone forgets to assign the value the transaction will be marked as succeeded. Not sure if really relevant or important but wanted to share a thought.

Reply:
yea.. I thought about this as well when setting this up. here was my thought process:
* In most cases, it's better to incorrectly think a tx succeeded than failed (don't transfer funds twice)
* Having a field name `Succeeded` or similar could be confusing since we don't know what the caller considers success, just that the tx didn't fail with an error. So it's more precise to signal that it failed.

Reply:
Yeah was thinking similarly about succeeded it's very ambigious. Good point on failed false by default is safer, I was thinking the other way around but it makes sense yeah. I just wanted to raise this in case it makes sense so good you have thought about it :)